### ISSUE TITLE
workgroup_3_2024_I

### ISSUE THREAD

**Issue by alexanderquispe** [d2cml-ai/CausalAI-Course#932]
# Neyman Orthogonality Proof
* Using Markdown cells, You have to prove the Neyman Orthogonality Condition. Please, first **read** this [Neyman_orthogonality_Proof](https://github.com/alexanderquispe/ECO224/blob/main/Lectures/Lecture_5/Neyman%20Orthogonality_proof.pdf) as tutorial.
* You have to prove that from  the next equation

$$y=\alpha D+\beta^{\prime} W+\epsilon $$

$$ \tilde{y}=y-X^{\prime}  W; \quad  \tilde{D}=D-X_{n 1.1} W$$

and 
$$\alpha=\underset{a \in R}{\arg min}E\left[	\tilde y - a \tilde D\right] ^{\prime} = E[\tilde {D ^ 2}]^{-1} E[\tilde {D}y] $$

you can get the next expression (Use the Implicit Theorem Fucntion)
$$\quad \frac{d a}{\partial \eta}=- \underbrace{\left[\frac{\partial M}{\partial a}\left(\alpha, \eta_0\right)\right]} _{S_1}^{-1} \underbrace{\left[\frac{\partial M}{\partial \eta} \left(\alpha, \eta^0\right)\right]} _{S_2} $$

Then, You will have to show that $S1=0$ and $S2=0$

# Code Section 

## Orthogonal Learning
You need to replicate the notebook-experiment-on-orthogonal-learning. 
Here are the scripts in [Python](https://github.com/alexanderquispe/CausalAI-Course/blob/main/labs/python_notebooks/Lab4/python_notebook_experiment_on_orthogonal_learning.ipynb), [R](https://github.com/alexanderquispe/CausalAI-Course/blob/main/labs/r_notebooks/Lab4/r_experiment_non_orthogonal_irnb.ipynb) and [Julia ](https://github.com/alexanderquispe/CausalAI-Course/blob/main/labs/julia_notebooks/Lab4/09_Julia_OrthogonalLearning_lab4.ipynb)

Based on those scripts, you will replicate Simulation Design 1 three times using different trials. So you have to use
- The equation of Y for the three cases now is: Y = 10\*D + beta\*X + rnorm(n)
-  B = 100, B = 1000, B = 10000. 
- Explain what you observe in the distribution of the histograms. Where the orthogonal and Naive should be centered? 
- Why does this happen? Give an econometric explanation of this. 
- Label your axes and legends correctly in your figures. 
- The analysis should be done in Python, R, and Julia
- **You have to implement parallel computing for the three scripts. In R you can use [paralell](https://bookdown.org/rdpeng/rprogdatascience/parallel-computation.html), in python [multiprocess](https://towardsdatascience.com/multiprocessing-in-python-9d498b1029ca), for julia [Multi-Threading](https://www.youtube.com/watch?v=Aa33gSmd5gc)** 
- Explain why using parallel computing, the running time is lower. To get all points in this section you can use one example (maybe with B=100) using and without using parallel computing and calculate the running time for each. 

## Double Lasso - Using School data
* You have to create your analysis based on the paper 
[The Impact of High School Financial Education: Evidence from a Large-Scale Evaluation in Brazil](https://papers.ssrn.com/sol3/papers.cfm?abstract_id=2367083) in Python, R and Julia. 
* The data is available [here](https://github.com/gsbDBI/ExperimentData/tree/master/School) 

Run the next approaches (Feel free to use any specification. You can use one or more outcomes presented in the paper and also use any control variables you think are relevant):

- OLS
- Double LASSO using cross Validation ([GLMNET ](https://glmnet.stanford.edu/articles/glmnet.html)in R and [Sklearn ](https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LassoCV.html)in Python, [Lasso.jl](https://github.com/JuliaStats/Lasso.jl) in Julia. Or you can use any other library of your preference
- Double Lasso using theoretical Lambda (HDM package in R,Python and Julia)
- Double Lasso using **method="partialling out"**, this is a direct way to get results from the HDM package, check the Python script for an example, and also you have to read the [documentation](https://arxiv.org/pdf/1603.01700.pdf)  for more details. 
- Plot the main coefficient and its confidence interval from these 4 cases in one figure. Explain what are the patterns you found

**Deadline: 05/04/24at 23:59**
